package/cloudshell/cm/customscript/domain/script_downloader.py

```diff
@@ -23,53 +25,100 @@ def __init__(self, logger, cancel_sampler):
         :type cancel_sampler: CancellationSampler
         """
         self.logger = logger
-        self.cancel_sampler = cancel_sampler
-        self.filename_pattern = r'(?P<filename>^.*\.(sh|bash|ps1)$)'
+        self.cancel_sampler = cancel_sampler        
+        self.filename_pattern = r"(?P<filename>^.*\.?[^/\\&\?]+\.(sh|bash|ps1)(?=([\?&].*$|$)))"
```

Comment:
what does this change mean?

Reply:
the old regex could only handle the filename if it was in the end of the url, the new on can also handle the case where a token is added to the url, which happens sometimes when using a url from a private repo. like this - \bash.sh?token=asdasda
the new regex can extract the filename from that.